File: Python/Prodi.py
from Dosen import Dosen
from Mahasiswa import Mahasiswa

class Prodi():
    def __init__(self, nama_prodi="-", kode="-", listCourse=[], listDosen=[], listMahasiswa=[]):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__listCourse = listCourse
        self.__listDosen = listDosen
        self.__listMahasiswa = listMahasiswa
    
    def setNama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def getNama_prodi(self):
        return self.__nama_prodi

    def setKode(self, kode):
        self.__kode = kode

    def getKode(self):
        return self.__kode

    def setListCourse(self, listCourse):
        self.__listCourse = listCourse

    def getListCourse(self):
        return self.__listCourse
    
    def addListCourse(self, course):
        self.__listCourse.append(course)
    
    def getStrListCourse(self):
        return ', '.join(str(e.getNama_matakuliah()) for e in self.getListCourse() )

    def setListDosen(self, listDosen):
        self.__listDosen = listDosen

    def getListDosen(self):
        return self.__listDosen
    
    def addListDosen(self, dosen):
        self.__listDosen.append(dosen)
    
    def getStrListDosen(self):
        return ', '.join(str(e.getNama()) for e in self.getListDosen() )
	
    def setListMahasiswa(self, listMahasiswa):
        self.__listMahasiswa = listMahasiswa
	
    def getListMahasiswa(self):
        return self.__listMahasiswa
    
    def addListMahasiswa(self, mahasiswa):
        self.__listMahasiswa.append(mahasiswa)
    
    def getStrListMahasiswa(self):
        return ', '.join(str(e.getNama()) for e in self.getListMahasiswa() )

File: cpp/Prodi.cpp
// prodi memiliki list objek course, dosen dan mahasiswa

class Prodi{
	private:
		string nama_prodi;
		string kode;
		vector<Course> listCourse;
		vector<Dosen> listDosen;
		vector<Mahasiswa> listMahasiswa;
	public:
		Prodi(){
			this->nama_prodi = "-";
			this->kode = "-";
		}
		Prodi(string nama_prodi, string kode, vector<Course> listCourse, vector<Dosen> listDosen, vector<Mahasiswa> listMahasiswa){
			this->nama_prodi = nama_prodi;
			this->kode = kode;
			this->listCourse = listCourse;
			this->listDosen = listDosen;
			this->listMahasiswa = listMahasiswa;
		}
		void setNama_prodi(string nama_prodi){
			this->nama_prodi = nama_prodi;
		}
		string getNama_prodi(){
			return this->nama_prodi;
		}
		void setKode(string kode){
			this->kode = kode;
		}
		string getKode(){
			return this->kode;
		}
		void setListCourse(vector<Course> listCourse){
			this->listCourse = listCourse;
		}
		vector<Course> getListCourse(){
			return this->listCourse;
		}
		void addListCourse(Course course){
			this->listCourse.push_back(course);
		}
		string getStrListCourse(){		//mengembalikan nama nama matakuliah pada listCourse
			string course;
			for(vector<Course>::iterator ptr=listCourse.begin(); ptr<listCourse.end(); ptr++){
				course.append((*ptr).getNama_matakuliah());		//semua nama matakuliah disimpan pada string course
				if(ptr+1 != listCourse.end()){
					course.append(", ");
				}
			}
			return course;
		}
		void setListDosen(vector<Dosen> listDosen){
			this->listDosen = listDosen;
		}
		vector<Dosen> getListDosen(){
			return this->listDosen;
		}
		void addListDosen(Dosen dosen){
			this->listDosen.push_back(dosen);
		}
		string getStrListDosen(){		//mengembalikan nama nama dosen pada listDosen
			string dosen;
			for(vector<Dosen>::iterator ptr=listDosen.begin(); ptr<listDosen.end(); ptr++){
				dosen.append((*ptr).getNama());		//semua nama dosen disimpan pada string dosen
				if(ptr+1 != listDosen.end()){
					dosen.append(", ");
				}
			}
			return dosen;
		}
		void setListMahasiswa(vector<Mahasiswa> listMahasiswa){
			this->listMahasiswa = listMahasiswa;
		}
		vector<Mahasiswa> getListMahasiswa(){
			return this->listMahasiswa;
		}
		void addListMahasiswa(Mahasiswa mahasiswa){
			this->listMahasiswa.push_back(mahasiswa);
		}
		string getStrListMahasiswa(){		//mengembalikan nama nama mahasiswa pada listDosen
			string mhs;
			for(vector<Mahasiswa>::iterator ptr=listMahasiswa.begin(); ptr<listMahasiswa.end(); ptr++){
				mhs.append((*ptr).getNama());		//semua nama mahasiswa disimpan pada string mhs
				if(ptr+1 != listMahasiswa.end()){
					mhs.append(", ");
				}
			}
			return mhs;
		}
};
